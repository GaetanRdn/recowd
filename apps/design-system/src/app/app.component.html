<recowd-nx-welcome></recowd-nx-welcome>
