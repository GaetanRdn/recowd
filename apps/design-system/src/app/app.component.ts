import { Component } from '@angular/core';

@Component({
  selector: 'recowd-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss'],
})
export class AppComponent {
  title = 'design-system';
}
