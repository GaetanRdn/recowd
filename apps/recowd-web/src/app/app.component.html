<rc-rw-toolbar
  (partnersClick)="redirectTo('/partners')"
  (experienceClick)="redirectTo('/experiences')"
  (educationClick)="redirectTo('/degrees')"
  (contactClick)="redirectTo('/contact')"
></rc-rw-toolbar>

<div [@routeAnimations]="prepareRoute(outlet)">
  <router-outlet #outlet="outlet"></router-outlet>
</div>

<rc-rw-footer></rc-rw-footer>
