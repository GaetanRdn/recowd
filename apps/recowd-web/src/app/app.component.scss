:host {
  background-color: #333333;
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: min-content /* router-outlet = 0 */ 0 1fr min-content;
  min-height: 100vh;
  width: 100vw;
}
