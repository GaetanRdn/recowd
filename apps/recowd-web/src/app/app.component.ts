import { ChangeDetectionStrategy, Component, inject } from '@angular/core';
import { Router } from '@angular/router';

@Component({
  selector: 'rc-recowd-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class AppComponent {
  private readonly _router: Router = inject(Router);

  public redirectTo(route: string): void {
    this._router.navigateByUrl(route);
  }
}
