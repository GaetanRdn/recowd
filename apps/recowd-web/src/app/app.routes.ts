import { EducationPageComponent } from '../pages/education-page/education-page.component';
import { Routes } from '@angular/router';
import { PartnersPageComponent } from '../pages/partners-page/partners-page.component';
import { ExperiencesPageComponent } from '../pages/experiences-page/experiences-page.component';
import { ContactPageComponent } from '../pages/contact-page/contact-page.component';

export const routes: Routes = [
  {
    path: '',
    redirectTo: 'partners',
    pathMatch: 'full',
  },
  {
    path: 'degrees',
    component: EducationPageComponent,
    data: { animation: 'degrees' },
  },
  {
    path: 'partners',
    component: PartnersPageComponent,
    data: { animation: 'partners' },
  },
  {
    path: 'experiences',
    component: ExperiencesPageComponent,
    data: { animation: 'experiences' },
  },
  {
    path: 'contact',
    component: ContactPageComponent,
    data: { animation: 'contact' },
  },
  {
    path: '**',
    redirectTo: '/',
  },
];
