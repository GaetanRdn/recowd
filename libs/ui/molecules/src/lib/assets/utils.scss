@function map-get-deep($map, $level, $subLevel) {
  @return map-get($map, $level, $subLevel);
}

@mixin use-typography($typography) {
  @each $key in font-size, font-family, font-weight, line-height {
    #{$key}: map-get($typography, $key);
  }
}
