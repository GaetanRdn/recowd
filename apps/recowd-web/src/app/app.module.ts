import { inject, NgModule } from '@angular/core';
import { AppComponent } from './app.component';
import { HttpClient, HttpClientModule } from '@angular/common/http';
import { RouterModule } from '@angular/router';
import { RwFooterComponent, RwToolbarComponent } from '@recowd/ui/organisms';
import { BrowserModule } from '@angular/platform-browser';
import { routes } from './app.routes';
import {
  TranslateLoader,
  TranslateModule,
  TranslateService,
} from '@ngx-translate/core';
import { TranslateHttpLoader } from '@ngx-translate/http-loader';

export function createTranslateLoader(http: HttpClient) {
  return new TranslateHttpLoader(http, './assets/i18n/', '.json');
}

@NgModule({
  declarations: [AppComponent],
  imports: [
    BrowserModule,
    HttpClientModule,
    RouterModule.forRoot(routes, { initialNavigation: 'enabledBlocking' }),
    TranslateModule.forRoot({
      defaultLanguage: 'fr',
      loader: {
        provide: TranslateLoader,
        useFactory: createTranslateLoader,
        deps: [HttpClient],
      },
    }),
    RwToolbarComponent,
    RwFooterComponent,
  ],
  bootstrap: [AppComponent],
})
export class AppModule {
  private _translateService: TranslateService = inject(TranslateService);

  constructor() {
    this._translateService.use(navigator.language === 'fr' ? 'fr' : 'en');
  }
}
