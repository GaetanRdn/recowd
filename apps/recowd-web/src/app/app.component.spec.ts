import { TestBed, waitForAsync } from '@angular/core/testing';
import { AppComponent } from './app.component';
import { RouterTestingModule } from '@angular/router/testing';
import { TemplateLookup } from '@recowd/test/utils';

describe('AppComponent', () => {
  let templateLookup: TemplateLookup<AppComponent>;

  beforeEach(waitForAsync(() => {
    TestBed.configureTestingModule({
      imports: [RouterTestingModule, AppComponent],
    }).compileComponents();

    templateLookup = new TemplateLookup(AppComponent);
    templateLookup.detectChanges();
  }));

  it('should create the app', () => {
    expect(templateLookup.fixture.nativeElement.children).toMatchSnapshot();
  });
});
