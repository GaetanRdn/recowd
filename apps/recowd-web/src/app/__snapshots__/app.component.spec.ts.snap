// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should create the app 1`] = `
HTMLCollection [
  <rc-rw-toolbar>
    <rc-toolbar>
      <rc-rw-avatar>
        <img
          alt="avatar"
          class="rc-avatar"
          rcavatar=""
          src="assets/venom.png"
        />
      </rc-rw-avatar>
      <rc-education-button-with-icon>
        <a
          class="rc-button"
          rcbutton=""
          role="button"
        >
          <rc-material-icon
            class="material-icons-outlined"
            type="outlined"
          >
            school
          </rc-material-icon>
           degrees

        </a>
      </rc-education-button-with-icon>
      <rc-experience-button-with-icon>
        <a
          class="rc-button"
          rcbutton=""
          role="button"
        >
          <rc-material-icon
            class="material-icons-outlined"
            type="outlined"
          >
            history_edu
          </rc-material-icon>
           experiences

        </a>
      </rc-experience-button-with-icon>
      <rc-contact-button-icon>
        <a
          class="rc-button"
          rcbutton=""
          role="button"
        >
          <rc-material-icon
            class="material-icons-outlined"
            type="outlined"
          >
            mail
          </rc-material-icon>
        </a>
      </rc-contact-button-icon>
    </rc-toolbar>
  </rc-rw-toolbar>,
  <router-outlet />,
  <rc-rw-footer>
    <footer>
      <rc-toolbar>
        <rc-nx-custom-icon>
          <rc-custom-icon>
            <object
              data="assets/images/nx.svg"
              height="24"
              type="image/svg+xml"
              width="24"
            >
              <img
                alt="assets/images/nx"
                src="assets/images/nx.png"
              />
            </object>
          </rc-custom-icon>
        </rc-nx-custom-icon>
        <rc-angular-custom-icon>
          <rc-custom-icon>
            <object
              data="assets/images/angular.svg"
              height="24"
              type="image/svg+xml"
              width="24"
            >
              <img
                alt="assets/images/angular"
                src="assets/images/angular.png"
              />
            </object>
          </rc-custom-icon>
        </rc-angular-custom-icon>
        <rc-jest-custom-icon>
          <rc-custom-icon>
            <object
              data="assets/images/jest.svg"
              height="24"
              type="image/svg+xml"
              width="24"
            >
              <img
                alt="assets/images/jest"
                src="assets/images/jest.png"
              />
            </object>
          </rc-custom-icon>
        </rc-jest-custom-icon>
        <rc-storybook-custom-icon>
          <rc-custom-icon>
            <object
              data="assets/images/storybook.svg"
              height="24"
              type="image/svg+xml"
              width="24"
            >
              <img
                alt="assets/images/storybook"
                src="assets/images/storybook.png"
              />
            </object>
          </rc-custom-icon>
        </rc-storybook-custom-icon>
        <rc-sass-custom-icon>
          <rc-custom-icon>
            <object
              data="assets/images/sass.svg"
              height="24"
              type="image/svg+xml"
              width="24"
            >
              <img
                alt="assets/images/sass"
                src="assets/images/sass.png"
              />
            </object>
          </rc-custom-icon>
        </rc-sass-custom-icon>
        <rc-cypress-custom-icon>
          <rc-custom-icon>
            <object
              data="assets/images/cypress.svg"
              height="24"
              type="image/svg+xml"
              width="24"
            >
              <img
                alt="assets/images/cypress"
                src="assets/images/cypress.png"
              />
            </object>
          </rc-custom-icon>
        </rc-cypress-custom-icon>
      </rc-toolbar>
    </footer>
  </rc-rw-footer>,
]
`;
